### PR TITLE
[5.2] Added URL::has()

### PR DESCRIPTION
**Pull request by ctf0**
just a very tiny addition to make it even more easier to check for a specific segment/keyword in the url.

### CODE REVIEW

File: src/Illuminate/Routing/UrlGenerator.php [hunk 1]
```diff
@@ -128,6 +128,17 @@ public function current()
...
 
     /**
+     * Check if the current url have a specific keyword.
+     *
+     * @param  string $keyword
+     * @return true|false
+     */
+    public function has($keyword)
+    {
+        return strpos($this->current(), $keyword);
```
Comment (rap2hpoutre) [inline src/Illuminate/Routing/UrlGenerator.php]: Quoting the php doc about strpos:
> This function may return Boolean FALSE, but may also return a non-Boolean value which evaluates to FALSE. Please read the section on Booleans for more information. Use the === operator for testing the return value of this function.

So, if you want to return a boolean for your `has` method, you have to do something like:

```
 return strpos($this->current(), $keyword) !== false;
```

**Comment by GrahamCampbell** [comment]
Yes, this needs fixing please.

File: src/Illuminate/Routing/UrlGenerator.php [hunk 2]
```diff
@@ -128,6 +128,17 @@ public function current()
     }
 
     /**
+     * Check if the current url have a specific keyword.
+     *
+     * @param  string $keyword
+     * @return true|false
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/UrlGenerator.php]: bool please

File: src/Illuminate/Routing/UrlGenerator.php (hunk 1 above)
Comment (ctf0) [inline src/Illuminate/Routing/UrlGenerator.php]: thanx for the headsup :muscle:  , done

File: src/Illuminate/Routing/UrlGenerator.php [hunk 3]
```diff
@@ -128,6 +128,17 @@ public function current()
     }
 
     /**
+     * Check if the current url have a specific keyword.
+     *
+     * @param  string $keyword
```
Comment (themsaid) [inline src/Illuminate/Routing/UrlGenerator.php]: Separate between type & argument by 2 spaces.

```
@param  string(  )$keyword
```

**Comment by taylorotwell** [comment]
Would suggest using Str::contains(URL::current()...

**Comment by ctf0** [comment]
@taylorotwell the reason behind the URL::has() was using

```php
@if (URL::has('somthing'))

// instead of

@if (Str::contains(URL::current(), 'somthing'))
```
which is easier to remember and less code footprint :).

**Comment by vlakoff** [comment]
URL::macro('has', function ($needle) {
        return Str::contains($this->current(), $needle);
    });

Done. :information_desk_person:

**Pull request closed without merging**